modules/internal/ChapelArray.chpl

```diff
@@ -674,6 +674,11 @@ module ChapelArray {
     return y.type;
   }
 
+  proc chpl__buildDistType(type t: record) type {
+    compilerWarning("Using 'dmap' with this distribution type is deprecated; simply use '<DistName>(args)'");
```

Comment:
Oh, that's true and should be easy, huh?

Reply:
Hmm, this turned out not to be as easy / clean as I'd hoped.  Specifically, using `t:string`, results in the fully qualified type `Block(...)` signature rather than just the name `Block`, such that my first attempt in updating the warnings causes code like:

```chapel
const Dist: dmap(Block(1)) = new Block({1..10});
var B = new dmap(new Block({1..n}));
```

to generate warnings like:

```bash
warning: Using 'dmap' with the 'Block(1,int(64),unmanaged DefaultDist)' distribution type is deprecated; simply use 'Block(1,int(64),unmanaged DefaultDist)(args)'
warning: Using 'dmap' with the 'Block(1,int(64),unmanaged DefaultDist)' distribution type is deprecated; simply use 'new Block(1,int(64),unmanaged DefaultDist)(args)'
```

This feels non-helpful to me in that we wouldn't actually suggest anyone write the suggestions.  A midway point would be to only print the type signature in the first occurrence, resulting in messages like:

```bash
warning: Using 'dmap' with the 'Block(1,int(64),unmanaged DefaultDist)' distribution type is deprecated; please replace 'dmap(<DistName>(args))' with '<DistName>(args)'
warning: Using 'dmap' with the 'Block(1,int(64),unmanaged DefaultDist)' distribution type is deprecated; please replace 'new dmap(<DistName>(args))' with 'new <DistName>(args)'
```

but I'm still not sure that's a net win since I think people will rarely use the full type or even know what that third argument is (the sparse layout field).

@e-kayrakli : Thoughts?  Or do you know of a way to trim the type signature to just its name as a param expression?  (I didn't see any param string expressions that looked like they'd apply).

Reply:
Ugh, yeah, that's unfortunate. I don't think we have that much compile-time capability on our strings. Let's keep things as-is.

Could be a good addition to... Reflection module maybe? What do you think?

Reply:
> Let's keep things as-is.

OK.  Without trying to print the name of the type, I may adapt the second part of the wording from my hybrid solution (e.g., "please replace 'dmap(<DistName>(args))' with '<DistName>(args)'") because I think it may be crisper than what I currently have.  Let me look at it once more with clear eyes.

> Reflection module maybe? What do you think?

Possibly, yes.  Or a method on types (which we don't really have the capability to write today...).